indico/modules/events/editing/client/js/components/Timeline/reducer.js

```diff
@@ -0,0 +1,19 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {SET_LOADING, SET_DETAILS} from './actions';
+
+export default function reducer(state, action) {
+  switch (action.type) {
+    case SET_LOADING:
+      return {...state, isLoading: action.loading};
+    case SET_DETAILS:
+      return {...state, details: action.details};
```

Comment:
shouldn't this imply changing `isLoading` to `false`?

Reply:
Yeah, but in case of an error we would still have to dispatch a separate action to disable loading, thus we just send an action to set `loading` to `false` in the `finally` block.
```javascript

      try {
        const {data} = await indicoAxios.get(
          editableDetailsURL({confId: eventId, contrib_id: contributionId, type})
        );
        dispatch(actions.setDetails(camelizeKeys(data)));
      } catch (error) {
        handleAxiosError(error, false, true);
      } finally {
        dispatch(actions.setLoading(false));
      }
```

Reply:
sure, just wondered if it'd be better to only dispatch the explicit action in case of an error. anyway, i don't think there's a problem with the separate dispatches